ecc/bls12-377/fr/mimc/mimc.go

```diff
@@ -124,14 +123,14 @@ func (d *digest) checksum() fr.Element {
 	// Write guarantees len(data) % BlockSize == 0
 
 	// TODO @ThomasPiellard shouldn't Sum() returns an error if there is no data?
-	if len(d.data) == 0 {
+	// TODO: @Tabaie, @Thomas Piellard Now sure what to make of this
```

Comment:
2 solutions I guess; 
1. since `checksum` is called in `Sum` and `Sum` can return an error, we could do that if there is no data to hash.
2. refer to other Hash impl in Go std and see how Sum behaves with empty buffer.

Reply:
Took a look at SHA256, turns out for conventional hashes hashing nothing makes sense, and provides a valid input. I think our choice is between returning 0 or throwing an error.